tests/test_dataframe_regression.py

```diff
@@ -0,0 +1,259 @@
+    assert expected in obtained_error_msg
+
+    # Assertion error case 2: More than one invalid data
+    data1 = 1.1 * np.ones(5000)
+    data2 = 2.2 * np.ones(5000)
+    data1[500] += 0.1
+    data1[600] += 0.2
+    data2[700] += 0.3
+    with pytest.raises(AssertionError) as excinfo:
+        dataframe_regression.check(
+            pd.DataFrame.from_dict({"data1": data1, "data2": data2})
+        )
+    obtained_error_msg = str(excinfo.value)
+    expected = "\n".join(
+        [
+            "Values are not sufficiently close.",
+            "To update values, use --force-regen option.",
+        ]
+    )
+    assert expected in obtained_error_msg
+    expected = "\n".join(
+        [
+            "data1:",
+            "          obtained_data1       expected_data1                 diff",
+            "500  1.20000000000000018  1.10000000000000009  0.10000000000000009",
+            "600  1.30000000000000004  1.10000000000000009  0.19999999999999996",
+        ]
+    )
+    assert expected in obtained_error_msg
+    expected = "\n".join(
+        [
+            "data2:",
+            "     obtained_data2       expected_data2                 diff",
+            "700             2.5  2.20000000000000018  0.29999999999999982",
+        ]
+    )
+    assert expected in obtained_error_msg
+
+    # Assertion error case 3: More than one invalid data
+    data1 = 1.1 * np.ones(5000)
+    data2 = 2.2 * np.ones(5000)
+    data1[500] += 0.01
+    data2[500] += 0.01
+    with pytest.raises(AssertionError) as excinfo:
+        dataframe_regression.check(
+            pd.DataFrame.from_dict({"data1": data1, "data2": data2}),
+            tolerances={
+                "data1": dict(atol=1e-1, rtol=1e-17),
+                "data2": dict(atol=1e-17, rtol=1e-17),
+            },
+        )
+    obtained_error_msg = str(excinfo.value)
+    assert "  data1:" not in obtained_error_msg
+    assert (
+        "\n".join(
+            [
+                "Values are not sufficiently close.",
+                "To update values, use --force-regen option.",
+            ]
+        )
+        in obtained_error_msg
+    )
+    assert (
+        "\n".join(
+            [
+                "data2:",
+                "          obtained_data2       expected_data2                 diff",
+                "500  2.20999999999999996  2.20000000000000018  0.00999999999999979",
+            ]
+        )
+        in obtained_error_msg
+    )
+
+
+def test_different_data_types(dataframe_regression, no_regen):
+    data1 = np.ones(10)
+    # Smoke test: Should not raise any exception
+    dataframe_regression.check(pd.DataFrame.from_dict({"data1": data1}))
+
+    data2 = np.array([True] * 10)
+    with pytest.raises(
+        AssertionError,
+        match="Data type for data data1 of obtained and expected are not the same.",
+    ):
+        dataframe_regression.check(pd.DataFrame.from_dict({"data1": data2}))
+
+
+def test_n_dimensions(dataframe_regression, no_regen):
```

Comment:
Although I understand this is a consistent message, I find it a bit misleading, because in practice, we want to tell that n-dimensional arrays are not supported. I ask because on `num_regression`, the message is `Only 1D arrays are supported on num_data_regression fixture`.

Perhaps, at least, tell the user what he did wrong. Something like this:
`Only numeric data is supported on dataframe_regression fixture: Column "C" has unsupported data type "Object".`

What do you think?

Reply:
I ended up mixing things up in this test, sorry. Taking a second look to this I came to the fact that pandas does not accepts n-dimensional numpy arrays within dataframe columns.
So the assertion for 1D arrays inside `DataFrameRegressionFixture` will never fail. I think I can remove that assertion and rename this test as @tadeu  suggested

Reply:
sgtm